### DISCUSSION TITLE
重置表单，自定义组件没有被重置

### DISCUSSION THREAD

**Discussion by aaliny** [alibaba/formily#3649] [Q&A 2.x]
代码：https://codesandbox.io/s/busy-lucy-c4oozq?file=/App.tsx
描述：表单写了一个自定义组件，点击重置调用form.reset()需要清空表单项的值，但是自定义组件里的表单项由于通过state数据控制，所以不会被清空，是否可以参考antd form的重置让组件重新mount达到清空的功能，或者有什么其他的方法可以实现么

**Comment by aaliny** [comment]
知道可以将表单项的值直接作用到select的value上，以此监听，但是由于数据方的要求，我这个两级下拉框的组件最终需要拼接成一个值放到表单项中，所以只能先用state控制

**Comment by aaliny** [comment]
问题倒是用别的方法 解决了，但还是想问一下 为什么没有重新mount这个功能

**Reply by janryWang** [reply]
重置是内核行为，是对数据/状态的重置，不是对react组件的重置

**Reply by aaliny** [reply]
> 重置是内核行为，是对数据/状态的重置，不是对react组件的重置
那antd 为什么要重置组件呢
https://ant.design/components/form-cn#%E4%B8%BA%E4%BB%80%E4%B9%88-resetfields-%E4%BC%9A%E9%87%8D%E6%96%B0-mount-%E7%BB%84%E4%BB%B6
